### ISSUE TITLE
output mode should preserve line numbering

### ISSUE THREAD

**Issue by maxtaco** [maxtaco/tamejs#4]
There should be two output modes for the tamejs compiler, one that preserves line numbering so that the error messages line up, and one that nicely indents for debugging code output.  This is similar to what pubjs is doing now, can probably snarf some code from there, assuming we clean up the code inlining mess.

**Comment by felixge** [comment]
+1 - I'm very intrigued by this library, but not having line numbers match up makes practicing TDD very hard.

**Comment by maxtaco** [comment]
yeah, i agree.  i'm wavering as to how to handle this.  I've been told
that JS is working on source-code line translations:

https://bugs.webkit.org/show_bug.cgi?id=63940
https://bugzilla.mozilla.org/show_bug.cgi?id=618650

which would really help to solve this problem.....  not sure if i
should wait or not.

On Wed, Jul 27, 2011 at 1:32 AM, felixge
reply@reply.github.com
wrote:

**Comment by felixge** [comment]
Node 0.4 is frozen on V8 3.1, so even if V8 was to implement this right now, it would take months before it would be available for most folks. Now afaik v8 doesn't even have this on the roadmap yet, so we may be talking 6-12 month, if ever.

**Comment by maxtaco** [comment]
Sigh. Thanks for sleuthing that out....  Looks like
line-number-preservation it is.....

On Wed, Jul 27, 2011 at 9:01 AM, felixge
reply@reply.github.com
wrote:
